src/lib/Shop.js

```diff
@@ -339,74 +360,119 @@ class AutomationShop
+        }
 
-            if ((item.name == "Pokeball") || (item.name == "xAttack") || (item.name == "xClick"))
-            {
-                // Those basic items are always purchasable
-                itemData.isUnlocked = function() { return true; };
-                itemData.isPurchasable = itemData.isUnlocked;
-            }
-            else if (item.name == "Greatball")
+        for (const townName in TownList)
+        {
+            const town = TownList[townName];
+            for (const shop of town.content)
             {
-                itemData.isUnlocked = function() { return TownList["Lavender Town"].isUnlocked(); };
-                itemData.isPurchasable = function()
+                // We only want shops, so discard anything that is not a shop
+                if (shop.items === undefined)
+                {
+                    continue;
+                }
+                for (const item of shop.items)
+                {
+                    // Never add "items" that are
+                    // - Pokémons
+                    // - Berries
+                    // - Oak items
+                    // - Farm hands
+                    // - Key items
+                    if (item.image.includes("images/items/pokemonItem")
+                        || item.image.includes("images/items/berry")
+                        || item.image.includes("images/oakitems")
+                        || item.image.includes("images/profile")
+                        || item.image.includes("images/keyitems"))
                     {
-                        return this.__internal__isPokeMarkUnlocked()
-                            || (TownList["Lavender Town"].isUnlocked() && Automation.Utils.Route.canMoveToRegion(0))
-                            || TownList["Ecruteak City"].isUnlocked();
-                    }.bind(this);
-            }
-            else if (item.name == "Ultraball")
-            {
-                itemData.isUnlocked = function() { return TownList["Fuchsia City"].isUnlocked(); };
-                itemData.isPurchasable = function()
+                        continue;
+                    }
+
+                    // New item, so add it to the list
+                    if (sellableItems[item.name] === undefined)
                     {
-                        return this.__internal__isPokeMarkUnlocked()
-                            || (TownList["Fuchsia City"].isUnlocked() && Automation.Utils.Route.canMoveToRegion(0))
-                            || TownList["Blackthorn City"].isUnlocked();
-                    }.bind(this);
-            }
-            else if ((item.name == "Lucky_egg") || (item.name == "Token_collector"))
-            {
-                itemData.isUnlocked = function() { return TownList["Pewter City"].isUnlocked(); };
-                itemData.isPurchasable = function()
+                        sellableItems[item.name] = {item: item, towns: [townName], requirements: []};
```

Comment:
I would prefer if the towns list is empty here, and the push occurs all the time (not in an else)
It makes the code more readable IMO

Reply:
Ok, will change